src/main/java/org/assertj/core/condition/Mapped.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2021 the original author or authors.
+ */
+package org.assertj.core.condition;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import org.assertj.core.api.Condition;
+
+/**
+ * Container-<code>{@link Condition}</code> that does a mapping and then uses nested
+ * <code>{@link Condition}</code> to test the mapped actual value.
```

Comment:
add a complete code example that will show where `Mapped` is useful.

Reply:
Added examples